### PR TITLE
C# Testing

### PR DESCRIPTION
**Pull request by auroq**
Congrats on taking the first step to contributing to the [Sample Programs in Every Language][1] repository!

For simplicity, please make sure that your pull request includes one and only one sample program and language.

## Please Complete the Following

- [ ] I would **NOT** like credit for my contribution in the article (credit will be included otherwise)
- [ ] I plan to write the article for this code snippet
- [ ] I named the pull request using `<Sample Program> in <Language>` format
- [ ] I created/updated the language README
  - [ ] I added the sample program name to the README
  - [ ] I added fun facts (i.e. debut, developer, typing, etc.)
  - [ ] I added reference link(s) to the README
  - [ ] I added solution citations when necessary (see [plagiarism][2])
- [ ] I fixed #your-issue-number-here

## Notes

Feel free to put whatever you want here.

[1]: https://therenegadecoder.com/code/sample-programs-in-every-language/
[2]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/CONTRIBUTING.md#plagiarism

### CODE REVIEW

**Comment by auroq** [comment]
This is failing because it takes too long to download the mono container. Containers with C# sdks are really large... I'll look more into this tomorrow.

File: archive/c/c-sharp/testinfo.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+folder:
+  extension: ".cs"
+  naming: "pascal"
+
+container:
+  image: "mono"
+  tag: "5.18"  # Cannot use slim because it does not contain mcs for compilation
+  build: "mcs -reference:System.Numerics {{ source.name}}{{ source.extension }}"
```
Comment (jrg94) [inline archive/c/c-sharp/testinfo.yml]: I don't know anything about this, but is it possible to use `csc` to compile and `mono` to run as seen here: https://www.mono-project.com/docs/getting-started/mono-basics/. If so, that may loosen the requirement of having the complete image.

Comment (jrg94) [reply archive/c/c-sharp/testinfo.yml]: For future reference, builds timeout if they take longer than 10 minutes to produce log output. Also, public repos have a 50 minute limit on builds. We may run into some limit eventually that we'll have to try to circumvent.

Comment (auroq) [reply archive/c/c-sharp/testinfo.yml]: Mono is the c# compiler for mono. csc is the Roslyn compiler. They both now exist within mono, but we need the full image in order to use either. The slim image doesn't contain any sdk at all.

Comment (jrg94) [reply archive/c/c-sharp/testinfo.yml]: Haha fair enough! This is a tough one.

Comment (auroq) [reply archive/c/c-sharp/testinfo.yml]: I think, I'll just need to find a way to do log output while it's downloading the images.

Comment (jrg94) [reply archive/c/c-sharp/testinfo.yml]: Sounds like a plan. Too bad there isn't a nice workaround. It's fine locally, but downloading and building it every time in Travis is going to be time consuming.

**Pull request merged**